src/plugins/intel_gna/src/gna2_model_export_helper.cpp

```diff
@@ -147,34 +87,33 @@ std::string WriteAllEndpoints(std::ostream& outStream,
     return stream.str();
 }
 
-void WriteStringToTlv(std::ostream& outStream, const Gna2TlvType tlvType, const std::string& value) {
+static void WriteStringToTlv(std::ostream& outStream, const Gna2TlvType tlvType, const std::string& value) {
     const auto& valueTlv = GetStringAsTlv(tlvType, value);
     outStream.write(valueTlv.data(), valueTlv.size());
 }
 
-}  // namespace
-
 void ExportTlvModel(uint32_t modelId,
                     uint32_t deviceIndex,
                     std::ostream& outStream,
-                    std::string compileTarget,
+                    const DeviceVersion& compile_target,
                     const std::vector<GnaEndpoint>& allInputs,
                     const std::vector<GnaEndpoint>& allOutputs,
                     const GnaAllocations& allAllocations) {
-    const auto deviceVersionToExport = getTlvTargetFromCompileTarget(compileTarget);
+    if (compile_target == DeviceVersion::GNAEmbedded1_0) {
+        THROW_GNA_EXCEPTION << "Unsupported compile target for TLV export: GNA Embedded 1.0" << std::endl;
+    }
 
     uint32_t exportConfig;
     auto status = Gna2ModelExportConfigCreate(gnaUserAllocatorAlignedPage, &exportConfig);
     GNADeviceHelper::checkGna2Status(status, "Gna2ModelExportConfigCreate");
 
     status = Gna2ModelExportConfigSetSource(exportConfig, deviceIndex, modelId);
     GNADeviceHelper::checkGna2Status(status, "Gna2ModelExportConfigSetSource");
-    status = Gna2ModelExportConfigSetTarget(exportConfig, deviceVersionToExport);
+    status = Gna2ModelExportConfigSetTarget(exportConfig, static_cast<Gna2DeviceVersion>(compile_target));
```

Comment:
I would use coversion function from DeviceVersion to Gna2DeviceVersion - this are enums and values does not map 1:1

Reply:
Omitted by mistake.